MathLib/LinAlg/LinearSolver.h

```diff
@@ -20,50 +20,15 @@ class ConfigTree;
 namespace MathLib
 {
 
-//! Basic linear solver interface.
-template<typename Matrix, typename Vector>
-class LinearSolver
```

Comment:
If you kept the linear solver base class, we could in principle have EigenLinearSolver and EigenLISLinearSolver (and later on maybe others, too) in a single build.